backend/PolicyAdmin/PolicyConverter.cs

```diff
@@ -0,0 +1,269 @@
+            {
+                if (action == null)
+                {
+                    action = new List<string>();
+                }
+
+                action.Add(match.AttributeValue.Value);
+            }
+
+            return action;
+        }
+
+        private static List<string> GetRuleResources(XacmlAllOf allOf, List<string> resource)
+        {
+            foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.AbsoluteUri.Equals(XacmlConstants.MatchAttributeCategory.Resource)))
+            {
+                if (resource == null)
+                {
+                    resource = new List<string>();
+                }
+
+                resource.Add($"{match.AttributeDesignator.AttributeId.ToString()}:{match.AttributeValue.Value}");
+
+            }
+
+            return resource;
+        }
+
+        private static List<string> GetRuleSubjects(XacmlAllOf allOf, List<string> subject)
+        {
+            foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.AbsoluteUri.Equals(XacmlConstants.MatchAttributeCategory.Subject)))
+            {
+                if (subject == null)
+                {
+                    subject = new List<string>();
+                }
+
+                subject.Add($"{match.AttributeDesignator.AttributeId.ToString()}:{match.AttributeValue.Value}");
+            }
+
+            return subject;
+        }
+
+        public static XacmlPolicy ConvertPolicy(ResourcePolicy policyInput)
+        {
+
+            XacmlPolicy policyOutput = new XacmlPolicy(new Uri($"{AltinnXacmlConstants.Prefixes.PolicyId}{1}"), new Uri(XacmlConstants.CombiningAlgorithms.RuleDenyOverrides), new XacmlTarget(new List<XacmlAnyOf>()));
+
+            foreach (PolicyRule rule in policyInput.Rules)
+            {
+                policyOutput.Rules.Add(ConvertRule(rule));
+            }
+
+            policyOutput.ObligationExpressions.Add(GetAuthenticationLevelObligation(policyInput.RequiredAuthenticationLevelEndUser));
+
+            return policyOutput;
+        }
+
+
+        private static XacmlRule ConvertRule(PolicyRule policyRule)
+        {
+            XacmlRule xacmlRule = new XacmlRule(policyRule.RuleId, XacmlEffectType.Permit);
+            xacmlRule.Description = policyRule.Description;
+
+            List<XacmlAnyOf> ruleAnyOfs = new List<XacmlAnyOf>();
+            if (policyRule.Subject != null && policyRule.Subject.Count > 0)
+            {
+                ruleAnyOfs.Add(GetSubjectAnyOfs(policyRule.Subject));
+            }
+            if (policyRule.Resources != null && policyRule.Resources.Count > 0)
+            {
+                ruleAnyOfs.Add(GetResourceAnyOfs(policyRule.Resources));
+            }
+            if (policyRule.Actions != null && policyRule.Actions.Count > 0)
+            {
+                ruleAnyOfs.Add(GetActionAnyOfs(policyRule.Actions));
+            }
+
+            xacmlRule.Target = new XacmlTarget(ruleAnyOfs);
```

Comment:
What happens is the rule is incomplete here? e.g. missing subject/resource/action?

Reply:
Validation will handle that. We will convert anyway. Since we probably will be saving incomplete rules all the time